include/xtensor/xrandom.hpp

```diff
@@ -261,7 +261,16 @@ namespace xt
 
             if (de.dimension() == 1)
             {
-                std::shuffle(de.storage().begin(), de.storage().end(), engine);
+                auto first = de.storage().begin();
+                auto last = de.storage().end();
```

Comment:
We should take the chance to fix that problem from the previous version:

`storage().begin()` and `storage().end()` should be replaced with `de.begin()` and `de.end()`. The reason is we may want to apply this algorithm to expressions which do not provide a `storage` method, or for which this does not make sense (shuffling a view should not shuffle the whole underlying expression)

Reply:
Modified in the latest commit =)